### ISSUE TITLE
XML defined channel states looses pattern format

### ISSUE THREAD

**Issue by digitaldan** [eclipse/smarthome#967]
In the squeeze binding (and others) I have defined channels as being read only like so:
```
<state readOnly="true"></state>
```
Some time ago this worked fine.  Recently I have noticed that those channels would not show up in the UI, closer inspection in the rest api shows that they were missing the pattern property. If I remove the state tag or define a pattern within the state tag, then the item correctly shows a pattern property and shows its value correctly in the UI's.  I would expect that if a pattern is omitted that the item would be assigned the same pattern as if the whole state tag was omitted.  I have also not confirmed if this has anything to do with the item being readOnly.

**Comment by kaikreuzer** [comment]
> those channels would not show up in the UI

Which UI are you referring to exactly?

> the item would be assigned the same pattern as if the whole state tag was omitted

Which pattern is this then? The one defined in the label of an item file?

**Comment by digitaldan** [comment]
It was not showing up in the classic or paper ui.  The pattern was missing from the item, but now this just got more interesting.  I reverted back my changes in the channel xml to reproduce the issue.  Now I'm seeing different functionality, in my XML file I have defined:
```
    <channel-type id="title">
        <item-type>String</item-type>
        <label>Title</label>
        <description>Title of the current song</description>
        <state readOnly="true"></state>
    </channel-type>
```
but in the rest api I see
```
 "stateDescription": {
      "pattern": "%s",
      "readOnly": false,
      "options": []
    },
```
So now the pattern is showing up, but the "readOnly" property is incorrect.  I'm confused now since I'm getting different behavior, but according to https://www.eclipse.org/forums/index.php?t=msg&th=1074313&goto=1721782&#msg_1721782 I'm not completely losing my mind ;-)  I will try fiddling a bit more to see if I consistently can reproduce the issue.
